app/src/main/java/app/revanced/integrations/patches/LithoFilterPatch.java

```diff
@@ -214,7 +214,7 @@ public GeneralBytecodeAdsPatch() {
                 //"full_width_square_image_layout",
                 "video_display_full_buttoned_layout",
                 "_ad",
-                "ad_",
+                "|ad_",
```

Comment:
What about components called something_ad_something?

Reply:
You explicitly excluded download_button. But it's useless, because you can add a | on ```ad_```, to to identify ads components only instead of other components.

Reply:
That does not answer the question. The block rule `generalAds` will not consider components called something_ad.

Reply:
There are two ads whose name starts with ad_

So _ad was added as a valid name for both, avoiding adding two different names.

Reply:
And what components with the name "something_ad"?

Reply:
> And what components with the name "something_ad"?

Mhh...seems there is just one:

![Screenshot_20221113_190854_Samsung Internet](https://user-images.githubusercontent.com/108184954/201537359-3a4ebe99-b956-4774-96d6-412f096e9d30.png)

Reply:
So we need:

_ad
_ad_
ads_video_with_context